agent/tests/test_memcached.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "tlib_php.h"
+#include "tlib_datastore.h"
+
+#include "php_agent.h"
+#include "php_memcached.h"
+#include "util_system.h"
+
+tlib_parallel_info_t parallel_info
+    = {.suggested_nthreads = -1, .state_size = 0};
+
+static char* system_host_name;
+
+static void test_create_datastore_instance(void) {
+  /*
+   * Test : Normal operation.
+   */
+  assert_datastore_instance_equals_destroy(
+      "named socket",
+      &((nr_datastore_instance_t){
+          .host = system_host_name,
+          .database_name = "unknown",
+          .port_path_or_id = "/tmp/memcached.sock",
+      }),
+      nr_php_memcached_create_datastore_instance("/tmp/memcached.sock", 0));
+
+  assert_datastore_instance_equals_destroy(
+      "empty socket",
+      &((nr_datastore_instance_t){
+          .host = system_host_name,
+          .database_name = "unknown",
+          .port_path_or_id = "unknown",
+      }),
+      nr_php_memcached_create_datastore_instance("", 0));
+
+  assert_datastore_instance_equals_destroy(
+      "empty host",
+      &((nr_datastore_instance_t){
+          .host = "unknown",
+          .database_name = "unknown",
+          .port_path_or_id = "6379",
+      }),
+      nr_php_memcached_create_datastore_instance("", 6379));
+
+  assert_datastore_instance_equals_destroy(
+      "localhost socket",
+      &((nr_datastore_instance_t){
+          .host = system_host_name,
+          .database_name = "unknown",
+          .port_path_or_id = "localhost",
+      }),
+      nr_php_memcached_create_datastore_instance("localhost", 0));
```

Comment:
How is this test different from [this one](https://github.com/newrelic/newrelic-php-agent/blob/c40cbd42f74ca295924f55fbd6a9deab996b2085/agent/tests/test_memcached.c#L21-L28)? Use of `"localhost"` as the socket is confusing and it does not test any additional code paths than [this one](https://github.com/newrelic/newrelic-php-agent/blob/c40cbd42f74ca295924f55fbd6a9deab996b2085/agent/tests/test_memcached.c#L21-L28).

Reply:
organized and consolidated unit tests in [5597d7f](https://github.com/newrelic/newrelic-php-agent/pull/958/commits/5597d7fccf8657254c4c9d486612ef821b2754be)